### DISCUSSION TITLE
How to customized built-in Login view function?

### DISCUSSION THREAD

**Discussion by stevezkw1998** [pallets-eco/flask-security#1024] [General]
The API `/admin/login/` seems to be built-in in flask-security
However, I want to customize the login process in my project, how to implement it?

**Comment by jwag956** [comment]
Yes - Flask-Security registers views that provide the basic authentication requirements for your Flask application. In terms of customization - of course you can supply your own template (to change the style), you can sub-class the login form and provide any additional fields etc you need. The view isn't customizable - however - Flask-Security has a set of signals (documented here: https://flask-security.readthedocs.io/en/stable/api.html#signals) - the 'user_authenticated' signal is called upon successful user authentication which your application can use to process any additional information including any additional form inputs.

What does your application need specifically?
